src/main/webapp/app/shared/http/file.service.ts

```diff
@@ -11,15 +11,19 @@ export class FileService {
 
     /**
      * Fetches the template file for the given programming language
-     * @param {string} filename
      * @param {ProgrammingLanguage} language
      * @param {ProjectType} projectType (if available)
      * @returns json test file
      */
-    getTemplateFile(filename: string, language?: ProgrammingLanguage, projectType?: ProjectType) {
-        const languagePrefix = language ? `${language}/` : '';
-        const projectTypePrefix = projectType ? `${projectType}/` : '';
-        return this.http.get<string>(`${this.resourceUrl}/templates/${languagePrefix}${projectTypePrefix}${filename}`, { responseType: 'text' as 'json' });
+    getTemplateFile(language?: ProgrammingLanguage, projectType?: ProjectType) {
```

Comment:
Why is language optional here? On the server the language must be present.

Reply:
There is a README.md in the root of the templates.
As this PR doesn't intend to modify the existing behaviour, I left it like that. If someone want's to remove this because it's unused, I have no objections (other than please do it after this PR is merged xD)

Reply:
` public ResponseEntity<byte[]> getTemplateFile(@PathVariable ProgrammingLanguage language, @PathVariable Optional<ProjectType> projectType)`

If there is a root-readme it's unaccessible since the server endpoint does to allow the Programming language to be missing.

Reply:
True, I can actually adapt this in the client.